modules/appeals_api/app/workers/appeals_api/notice_of_disagreement_pdf_submit_job.rb

```diff
@@ -15,16 +15,48 @@ class NoticeOfDisagreementPdfSubmitJob
     def perform(notice_of_disagreement_id, retries = 0)
       @retries = retries
       stamped_pdf = generate_pdf(notice_of_disagreement_id)
+      upload_to_central_mail(notice_of_disagreement_id, stamped_pdf)
       File.delete(stamped_pdf) if File.exist?(stamped_pdf)
     end
 
     def generate_pdf(notice_of_disagreement_id)
       pdf_constructor = AppealsApi::NoticeOfDisagreementPdfConstructor.new(notice_of_disagreement_id)
       pdf_path = pdf_constructor.fill_pdf
-      # notice_of_disagreement.update!(status: 'submitting')
       notice_of_disagreement = NoticeOfDisagreement.find notice_of_disagreement_id
+      notice_of_disagreement.update!(status: 'submitting')
       inserted_text_pdf = pdf_constructor.insert_manual_fields(pdf_path)
       pdf_constructor.stamp_pdf(inserted_text_pdf, notice_of_disagreement.consumer_name)
     end
+
+    def upload_to_central_mail(notice_of_disagreement_id, pdf_path)
+      notice_of_disagreement = AppealsApi::NoticeOfDisagreement.find notice_of_disagreement_id
+      metadata = {
+        'veteranFirstName' => notice_of_disagreement.veteran_first_name,
+        'veteranLastName' => notice_of_disagreement.veteran_last_name,
+        'fileNumber' => notice_of_disagreement.file_number.presence || notice_of_disagreement.ssn,
+        'zipCode' => notice_of_disagreement.zip_code_5,
+        'source' => "Appeals-NOD-#{notice_of_disagreement.consumer_name}", # TODO: is consumer name a required field?
+        'uuid' => notice_of_disagreement.id,
+        'hashV' => Digest::SHA256.file(pdf_path).hexdigest,
+        'numberAttachments' => 0,
+        'receiveDt' => notice_of_disagreement.created_at.strftime('%Y-%m-%d %H:%M:%S'),
+        'numberPages' => PdfInfo::Metadata.read(pdf_path).pages,
+        'docType' => '01-0182' # TODO: need the correct format here
```

Comment:
Need to do discovery on these docTypes being sent to Central Mail (non-blocking). Why are they dashed? Why are the HLR and NOD `docType`s different lengths? Does it matter?

Reply:
Posted inquiry to banana peels public